stellar-multitenant/pkg/internal/provisioning/manager.go

```diff
@@ -89,6 +93,24 @@ func (m *Manager) ProvisionNewTenant(
 		return nil, fmt.Errorf("updating organization's name: %w", err)
 	}
 
+	// Provision distribution account for tenant if necessary
+	distributionAccPubKeys, err := m.distAccSigClient.BatchInsert(ctx, 1)
+	if err != nil {
+		if errors.Is(err, signing.ErrUnsupportedCommand) {
+			log.Ctx(ctx).Warnf(
+				"Account provisioning not needed for distribution account signature client type %s: %v",
+				m.distAccSigClient.Type(), err)
+		} else {
+			return nil, fmt.Errorf("provisioning distribution account: %w", err)
+		}
+	}
+	if len(distributionAccPubKeys) != 1 {
+		return nil, fmt.Errorf("expected single distribution account public key, got %d", len(distributionAccPubKeys))
+	}
+
+	distributionAccPubKey := distributionAccPubKeys[0]
+	log.Ctx(ctx).Infof("distribution account %s created for tenant %s", distributionAccPubKey, t.Name)
+
```

Comment:
I'll add support for deleting the keys from the tenant admin tb, and the relevant tss tb as well. I think we should really prioritize adding a tx wrapper around the entire provisioning block since any kind of errors will result in a tenant record w/ inconsistent state.

Reply:
Good call! I've created a ticket for that in https://stellarorg.atlassian.net/browse/SDP-1107